system-tests/maven-external-upload-test/src/test/java/com/teamscale/upload/MavenExternalUploadSystemTest.java

```diff
@@ -0,0 +1,79 @@
+package com.teamscale.upload;
+
+import com.teamscale.test.commons.ExternalReport;
+import com.teamscale.test.commons.SystemTestUtils;
+import com.teamscale.test.commons.TeamscaleMockServer;
+import org.conqat.lib.commons.io.ProcessUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+/**
+ * Runs several Maven projects' Surefire and Failsafe tests that produce coverage via the Jacoco Maven plugin
+ * Checks that the Jacoco reports are correctly uploaded to a Teamscale instance.
+ */
+public class MavenExternalUploadSystemTest {
+
+	/**
+	 * This port must match what is configured for the -javaagent line in the corresponding POM of the Maven test
+	 * project.
+	 */
+	private static final int FAKE_TEAMSCALE_PORT = 65432;
+
+	private static TeamscaleMockServer teamscaleMockServer = null;
+
+	private static final String NESTED_MAVEN_PROJECT_NAME = "nested-project";
+
+	private static final String FAILING_MAVEN_PROJECT_NAME = "failing-project";
+
+	@BeforeEach
+	public void startFakeTeamscaleServer() throws Exception {
+		if (teamscaleMockServer == null) {
+			teamscaleMockServer = new TeamscaleMockServer(FAKE_TEAMSCALE_PORT).acceptingReportUploads();
+		}
+		teamscaleMockServer.uploadedReports.clear();
+	}
+
+	private ProcessUtils.ExecutionResult runCoverageUploadGoal(String projectPath) {
+		File workingDirectory = new File(projectPath);
+		try {
+			return ProcessUtils.execute(new ProcessBuilder("./mvnw", "com.teamscale:teamscale-maven-plugin:upload-coverage").directory(workingDirectory));
+		} catch (IOException e) {
+			fail(String.valueOf(e));
+		}
+		return null;
```

Comment:
This method would probably fail on a windows machine (not in WSL) because of the `./mwnw` command. I would suggest to check the current os first and change the process builder command in the following way: 
```suggestion
        File workingDirectory = new File(projectPath);
	String executable = "./mvnw";
		if (SystemUtils.isWindows()) {
			executable = Paths.get(projectPath, "mvnw.cmd").toUri().getPath();
		}
		try {
	return ProcessUtils.execute(new ProcessBuilder(executable , MAVEN_COVERAGE_UPLOAD_GOAL).directory(workingDirectory));
	}
	catch (IOException e) {
			fail(String.valueOf(e));
		}
		return null;
```
I don't know why, but on my windows machine it was important to prefix the project path to the executable command, even when setting the `workingDirectory` with `.directory(workingDirectory)`

Also, please add a constant for "com.teamscale:teamscale-maven-plugin:upload-coverage"